eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -51,9 +52,13 @@ class MultiPartHandler(nodeParams: NodeParams, register: ActorRef, db: IncomingP
   /** Can be overridden to do custom post-processing on successfully received payments. */
   def postFulfill(paymentReceived: PaymentReceived)(implicit log: LoggingAdapter): Unit = ()
 
+  case class PaymentRequestResult(replyTo: ActorRef, res: Try[PaymentRequest])
+
   override def handle(implicit ctx: ActorContext, log: DiagnosticLoggingAdapter): Receive = {
     case ReceivePayment(amount_opt, desc, expirySeconds_opt, extraHops, fallbackAddress_opt, paymentPreimage_opt, paymentType) =>
-      Try {
+      implicit val ec: ExecutionContextExecutor = ctx.dispatcher
+      val replyTo = ctx.sender()
+      Future {
```

Comment:
I think it would be cleaner (more actor-like) to spawn a very simple actor for each payment request generation.

Reply:
Also, best would probably to put this in a separate PR, I'll do that.

Reply:
Reverted in 0672cdc.